llama_index/llms/sagemaker_llm_endpoint.py

```diff
@@ -0,0 +1,281 @@
+
+        def gen() -> CompletionResponseGen:
+            raw_text = ""
+            prev_clean_text = ""
+            for response in self._client.invoke_endpoint_with_response_stream(
+                EndpointName=self.endpoint_name,
+                Body=request_body,
+                ContentType=self.content_handler.content_type,
+                Accept=self.content_handler.accept,
+                **self.endpoint_kwargs,
+            )["Body"]:
+                delta = self.content_handler.deserialize_streaming_output(
+                    response["PayloadPart"]["Bytes"]
+                )
+                raw_text += delta
+                clean_text = self.content_handler.remove_prefix(raw_text, prompt)
+                delta = clean_text[len(prev_clean_text) :]
+                prev_clean_text = clean_text
+
+                yield CompletionResponse(text=clean_text, delta=delta, raw=response)
+
+        return gen()
+
+    @llm_chat_callback()
+    def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
+        prompt = self.messages_to_prompt(messages)
+        completion_response = self.complete(prompt, formatted=True, **kwargs)
+        return completion_response_to_chat_response(completion_response)
+
+    @llm_chat_callback()
+    def stream_chat(
+        self, messages: Sequence[ChatMessage], **kwargs: Any
+    ) -> ChatResponseGen:
+        prompt = self.messages_to_prompt(messages)
+        completion_response_gen = self.stream_complete(prompt, formatted=True, **kwargs)
+        return stream_completion_response_to_chat_response(completion_response_gen)
+
+    @llm_chat_callback()
+    async def achat(
+        self,
+        messages: Sequence[ChatMessage],
+        **kwargs: Any,
+    ) -> ChatResponse:
+        return self.chat(messages, **kwargs)
+
+    @llm_chat_callback()
+    async def astream_chat(
+        self,
+        messages: Sequence[ChatMessage],
+        **kwargs: Any,
+    ) -> ChatResponseAsyncGen:
+        async def gen() -> ChatResponseAsyncGen:
+            for message in self.stream_chat(messages, **kwargs):
+                yield message
+
+        # NOTE: convert generator to async generator
+        return gen()
+
+    @llm_completion_callback()
+    async def acomplete(
+        self, prompt: str, formatted: bool = False, **kwargs: Any
+    ) -> CompletionResponse:
+        return self.complete(prompt, formatted=formatted, **kwargs)
+
+    @llm_completion_callback()
+    async def astream_complete(
+        self, prompt: str, formatted: bool = False, **kwargs: Any
+    ) -> CompletionResponseAsyncGen:
+        async def gen() -> CompletionResponseAsyncGen:
+            for message in self.stream_complete(prompt, formatted=formatted, **kwargs):
+                yield message
+
+        # NOTE: convert generator to async generator
+        return gen()
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "sagemaker_endpoint"
```

Comment:
nit: can we do "sagemaker_llm_endpoint"?

Reply:
corrected!